sklearn/compose/_column_transformer.py

```diff
@@ -980,6 +985,11 @@ def fit_transform(self, X, y=None, **params):
             sum of n_components (output dimension) over transformers. If
             any result is a sparse matrix, everything will be converted to
             sparse matrices.
+
+        Notes
+        -----
+        If `X` is a sparse matrix and not row-wise indexable by default, it will
+        be internally converted to the CSR format for efficient processing.
```

Comment:
Efficiency depends on different use cases:

 - If the inner transformers works on the CSC directly, then it's most efficient for the column transformer to use CSC.
 - If the inner transformers works on CSR, then this PR is the most efficient.

I'm okay with this PR using CSR, but I would remove the "efficient processing" in the docstring:

```suggestion
        be internally converted to the CSR format.
```